File: 0075-sort-colors/0075-sort-colors.cpp
class Solution {
  void swap(int *a, int *b) {
    int temp = *a;
    *a = *b;
    *b = temp;
  }
  int Partition(vector<int> & arr, int front, int end) {
    int pivot = arr[end];
    int i = front - 1;
    for (int j = front; j < end; j++) {
      if (arr[j] < pivot) {
        i++;
        swap(&arr[i], &arr[j]);
      }
    }
    i++;
    swap(&arr[i], &arr[end]);
    return i;
  }
  void QuickSort(vector<int> &arr, int front, int end) {
    if (front < end) {
      int pivot = Partition(arr, front, end);
      QuickSort(arr, front, pivot - 1);
      QuickSort(arr, pivot + 1, end);
    }
  }

 public:
  void sortColors(vector<int> &nums) {
    QuickSort(nums, 0, nums.size() - 1);
  }
};

File: 0075-sort-colors/0075-sort-colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        curr = left = 0
        right = len(nums) - 1
        while curr <= right:
            if nums[curr] == 0:
                nums[left], nums[curr] = nums[curr], nums[left]
                left += 1
                curr += 1
            elif nums[curr] == 2:
                nums[right], nums[curr] = nums[curr], nums[right]
                right -= 1
            else:
                curr += 1
        
        return nums
        